### PR TITLE
kube play: update the handling of PersistentVolumeClaim

### LINKED ISSUE #16348: play kube: cannot restart pod with named volumes
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I am deploying the following YAML file via play kube:
```yaml
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: megamek-latest-savegames
  namespace: megamek-server-latest
spec:
  accessModes:
    - ReadWriteOnce
  storageClassName: local-path
  resources:
    requests:
      storage: 2Gi
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: megamek-latest-logs
  namespace: megamek-server-latest
spec:
  accessModes:
    - ReadWriteOnce
  storageClassName: local-path
  resources:
    requests:
      storage: 2Gi
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: megamek-latest-mmconf
  namespace: megamek-server-latest
spec:
  accessModes:
    - ReadWriteOnce
  storageClassName: local-path
  resources:
    requests:
      storage: 2Gi
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: megamek-server-latest
  namespace: megamek-server-latest
  labels:
    app: megamek-server-latest
spec:
  replicas: 1
  selector:
    matchLabels:
      app: megamek-server-latest
  template:
    metadata:
      labels:
        app: megamek-server-latest
    spec:
      volumes:
        - name: megamek-latest-savegames
          persistentVolumeClaim:
            claimName: megamek-latest-savegames
        - name: megamek-latest-mmconf
          persistentVolumeClaim:
            claimName: megamek-latest-mmconf
        - name: megamek-latest-logs
          persistentVolumeClaim:
            claimName: megamek-latest-logs
      containers:
        - name: megamek-server-latest
          image: quay.io/martjack/megamek-server:0.49.10
          ports:
            - containerPort: 2346
              name: megamek
          volumeMounts:
            - mountPath: "/home/jboss/megamek/savegames"
              name: megamek-latest-savegames
            - mountPath: "/home/jboss/megamek/logs"
              name: megamek-latest-logs
            - mountPath: "/home/jboss/megamek/mmconf"
              name: megamek-latest-mmconf
```
**Steps to reproduce the issue:**

1. podman play kube deploy-megamek.yaml

2. podman play kube deploy-megamek.yaml --down

3. podman play kube deploy-megamek.yaml

**Describe the results you received:**

In step 3, an error like the following occurs:
```
podman kube play deploy-megamek.yaml 
Error: volume with name megamek-latest-mmconf already exists: volume already exists
```

**Describe the results you expected:**

I expected the pod to re-start. It seems important that, for example, when re-starting a node running services via the systemd integration, that pods with volumes should be able to shutdown and restart gracefully

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Client:       Podman Engine
Version:      4.3.0
API Version:  4.3.0
Go Version:   go1.19.2
Built:        Fri Oct 21 03:09:51 2022
OS/Arch:      linux/amd64
```

**Output of `podman info`:**

```
host:
  arch: amd64
  buildahVersion: 1.28.0
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.4-3.fc37.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.4, commit: '
  cpuUtilization:
    idlePercent: 99.15
    systemPercent: 0.33
    userPercent: 0.52
  cpus: 2
  distribution:
    distribution: fedora
    version: "37"
  eventLogger: journald
  hostname: srv-f-podman.imladris.lan
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 344800018
      size: 1
    - container_id: 1
      host_id: 2147549184
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 344800018
      size: 1
    - container_id: 1
      host_id: 2147549184
      size: 65536
  kernel: 6.0.5-300.fc37.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 15353282560
  memTotal: 16772775936
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.6-2.fc37.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.6
      commit: 18cf2efbb8feb2b2f20e316520e0fd0b6c41ef4d
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +YAJL
  os: linux
  remoteSocket:
    path: /run/user/344800018/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-8.fc37.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 8589930496
  swapTotal: 8589930496
  uptime: 2h 25m 41.00s (Approximately 0.08 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/podman/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/podman/.local/share/containers/storage
  graphRootAllocated: 41872785408
  graphRootUsed: 2622685184
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 2
  runRoot: /run/user/344800018/containers
  volumePath: /home/podman/.local/share/containers/storage/volumes
version:
  APIVersion: 4.3.0
  Built: 1666339791
  BuiltTime: Fri Oct 21 03:09:51 2022
  GitCommit: ""
  GoVersion: go1.19.2
  Os: linux
  OsArch: linux/amd64
  Version: 4.3.0
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman` or `brew info podman`):**

```
podman-4.3.0-2.fc37.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

This is running in a KVM VM on a Fedora 37 hypervisor

### PR DESCRIPTION
**Pull request by ygalblum**
Up - do not fail if volume already exists, use the existing one
Down - allow the user to remove the volume by passing --force
Add tests

Signed-off-by: Ygal Blum <ygal.blum@gmail.com>

#### Does this PR introduce a user-facing change?

```release-note
kube play: update the handling of PersistentVolumeClaim - reuse existing volumes and allow their removal with down
```
Resolves #16348

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
Code LGTM.

Are you intending to use that in the `podman-kube@` template?  If so, we need to massage that a bit I guess.  `kube play --replace` does not remove volumes.  It may be worth adding a `StartPre` directive using `kube down`.

**Comment by vrothberg** [comment]
@baude PTAL

File: cmd/podman/kube/play.go [hunk 1]
```diff
@@ -138,6 +138,7 @@ func playFlags(cmd *cobra.Command) {
 	flags.BoolVarP(&playOptions.Quiet, "quiet", "q", false, "Suppress output information when pulling images")
 	flags.BoolVar(&playOptions.TLSVerifyCLI, "tls-verify", true, "Require HTTPS and verify certificates when contacting registries")
 	flags.BoolVar(&playOptions.StartCLI, "start", true, "Start the pod after creating it")
+	flags.BoolVar(&playOptions.Force, "force", false, "remove volumes")
```
Comment (mheon) [inline cmd/podman/kube/play.go:141]: Capitalize the R in remove please

Comment (mheon) [reply cmd/podman/kube/play.go:141]: And can you mention that this only works with `--down`?

Comment (mheon) [reply cmd/podman/kube/play.go:141]: Finally: can you add a check to error if `--force` is passed but `--down` is not?

File: docs/source/markdown/podman-kube-play.1.md.in [hunk 2]
```diff
@@ -138,6 +138,10 @@ Use *path* as the build context directory for each image. Requires --build optio
 
 @@option creds
 
+#### **--force**
+
+Tear down the volumes linked to the PersistentVolumeClaims as part --down
```
Comment (mheon) [inline docs/source/markdown/podman-kube-play.1.md.in:143]: "as part of `--down`."

**Comment by mheon** [comment]
Bunch of little things, overall looks fine

File: cmd/podman/kube/play.go (hunk 1 above)
Comment (ygalblum) [reply cmd/podman/kube/play.go]: Fixed

**Comment by rhatdan** [comment]
@umohnani8 PTAL

**Comment by umohnani8** [comment]
code LGTM

**Pull request merged**